tests/darwin/exporter/formats/export_mask_test.py

```diff
@@ -0,0 +1,581 @@
+        )
+
+
+# Test RLE decoder
+def test_rle_decoder() -> None:
+    predication = [1, 2, 3, 4, 5, 6]
+    expectation = [1, 1, 3, 3, 3, 3, 5, 5, 5, 5, 5, 5]
+
+    assert rle_decode(predication) == expectation
+
+    odd_number_of_integers = [1, 2, 3, 4, 5, 6, 7]
+    with pytest.raises(ValueError):
+        rle_decode(odd_number_of_integers)
+
+
+# Test render_polygons
+def test_render_polygons() -> None:
+    # Create some mock data for testing
+    mask = np.zeros((100, 100), dtype=np.uint8)
+    colours: dt.MaskTypes.ColoursDict = {}
+    categories: dt.MaskTypes.CategoryList = []
+    annotations: List[dt.AnnotationLike] = [
+        dt.Annotation(
+            dt.AnnotationClass("cat1", "polygon"),
+            {"path": [{"x": 10, "y": 10}, {"x": 20, "y": 10}, {"x": 20, "y": 20}, {"x": 10, "y": 20}]},
+        ),
+        dt.Annotation(
+            dt.AnnotationClass("cat2", "polygon"),
+            {"path": [{"x": 30, "y": 30}, {"x": 40, "y": 30}, {"x": 40, "y": 40}, {"x": 30, "y": 40}]},
+        ),
+        dt.Annotation(
+            dt.AnnotationClass("cat1", "polygon"),
+            {"path": [{"x": 50, "y": 50}, {"x": 60, "y": 50}, {"x": 60, "y": 60}, {"x": 50, "y": 60}]},
+        ),
+        dt.Annotation(
+            dt.AnnotationClass("cat1", "polygon"),
+            {"path": [{"x": 10, "y": 80}, {"x": 20, "y": 80}, {"x": 20, "y": 60}]},
+        ),
+        dt.Annotation(
+            dt.AnnotationClass("cat3", "complex_polygon"),
+            {
+                "paths": [
+                    [{"x": 70, "y": 70}, {"x": 80, "y": 70}, {"x": 80, "y": 80}, {"x": 70, "y": 80}],
+                    [{"x": 75, "y": 75}, {"x": 75, "y": 78}, {"x": 78, "y": 78}],
+                ]
+            },
+        ),
+    ]
+    annotation_file = dt.AnnotationFile(
+        Path("testfile"),
+        "testfile",
+        set([a.annotation_class for a in annotations]),
+        annotations,
+    )
+    height = 100
+    width = 100
+
+    # Call the function with the mock data
+    errors, new_mask, new_categories, new_colours = render_polygons(
+        mask, colours, categories, annotations, annotation_file, height, width
+    )
+
+    assert not errors
+
+    # Check that the mask was modified in place
+    assert_array_equal(mask, new_mask)  # type: ignore
+
+    # Check that the categories and colours were updated correctly
+    assert new_categories == ["cat1", "cat2", "cat3"]
+    assert new_colours == {"cat1": 1, "cat2": 2, "cat3": 3}
+
+    # Check that the polygons were drawn correctly
+
+    this_file_dir = Path(__file__).parent
+    expected_data_path = (this_file_dir / ".." / ".." / "data").resolve()
+    expected_mask: NDArray = np.array(np.fromfile(expected_data_path / "expected_mask.bin", dtype=np.uint8)).reshape(  # type: ignore
+        (100, 100)
+    )
+
+    assert_array_equal(new_mask, expected_mask)  # type: ignore
+
+
+# Test render_raster
+def test_render_raster() -> None:
+    rle_code: List[int] = [
```

Comment:
Thoughts on having a pytest conftest sort of structure with all these fixtures seperated from the tests? Make the actual test code more readable perhaps, open to discussion.

Reply:
I generally do that, yeah, but this doesn't currently - I think mainly because people have enclosed their fixtures in describe blocks.